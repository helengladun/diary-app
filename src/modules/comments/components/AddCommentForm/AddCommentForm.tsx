import React, {ChangeEvent, KeyboardEvent} from 'react';

interface IProps {
    submitHandler: (e: KeyboardEvent<HTMLTextAreaElement>) => void,
    changeHandler: (e: ChangeEvent<HTMLTextAreaElement>) => void
    value: string
}

export const AddCommentForm = (props: IProps) => (
    <form className="textarea-box">
        <div className="square textarea-box__colored-square"></div>
        <textarea
            className="textarea-box__textarea primary-textarea"
            value={props.value}
            onKeyUp={props.submitHandler}
            onChange={props.changeHandler}/>
    </form>
);