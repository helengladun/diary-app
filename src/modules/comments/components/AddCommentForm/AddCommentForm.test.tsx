import React from 'react';
import {shallow} from 'enzyme';
import {AddCommentForm} from "./AddCommentForm";

describe('AddCommentForm component', () => {
    const submitHandler = jest.fn();
    const changeHandler = jest.fn();
    const wrapper = shallow(<AddCommentForm
            value={'text'}
            submitHandler={submitHandler}
            changeHandler={changeHandler}
        />);

    test('should render AddCommentForm correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

