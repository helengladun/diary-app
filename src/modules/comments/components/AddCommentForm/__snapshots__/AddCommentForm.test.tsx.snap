// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCommentForm component should render CommentForm correctly 1`] = `
<form
  className="textarea-box"
>
  <div
    className="square textarea-box__colored-square"
  />
  <textarea
    className="textarea-box__textarea primary-textarea"
    onChange={[MockFunction]}
    onKeyUp={[MockFunction]}
    value="text"
  />
</form>
`;
